File: PuntoVB/Punto.vb
Public Class Punto
    Protected x1 As Integer
    Public Property X() As Integer
        Get
            Return x1
        End Get

        Set(ByVal value As Integer)
            x1 = value
        End Set
    End Property
    Public Sub New()
        x1 = 0

    End Sub
    Public Sub New(x As Integer)
        x1 = x

    End Sub
    Public Function Distancia(p As Punto) As Double
        Dim resultado As Double = (x1 - p.X)
        Return resultado
    End Function
    Public Overrides Function ToString() As String
        Return "(" + x1 + ")"
    End Function
End Class

File: PuntoC/Punto.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace PuntoC
{
    public class Punto
    {
        protected int _x;
        public int X
        {
            get { return _x; }
            set { _x = value; }
        }
        public Punto()
        {
            _x = 0;
        }

        public Punto(int x)
        {
            _x = x;
        }
        public override string ToString()
        {
            return "(" + _x + ")";
        }
        public double Distancia(Punto p)
        {
            double resultado = (_x - p.X);
            return resultado;
        }
    }
}
